runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -1129,3 +1129,159 @@ NNFW_STATUS nnfw_session::set_backends_per_operation(const char *backend_setting
 
   return NNFW_STATUS_NO_ERROR;
 }
+
+NNFW_STATUS nnfw_session::train_prepare(const nnfw_train_info *info)
+{
+  // We may need different state to represent training model is loaded
+  if (!isStateModelLoaded())
+  {
+    std::cerr << "Error during model prepare training: ";
+    if (_state == State::PREPARED_TRAINING)
+      std::cerr << "prepare should be run once";
+    else
+      std::cerr << "invalid state";
+    std::cerr << std::endl;
+    return NNFW_STATUS_INVALID_STATE;
+  }
+
+  // Use default info if info is null, otherwise use given info
+  (void)info;
```

Comment:
If the info is null, how about treating it as an error to clarify operation?

Reply:
It is designed to use training info in model metadata if `info` param is `nullptr`. Please refer API comment in `nnfw_experimental.h`.